src/main/java/com/suse/saltstack/netapi/client/SaltStackClient.java

```diff
@@ -44,15 +45,11 @@ public SaltStackClient(String url) throws SaltStackException {
      * @param connectionFactory Connection Factory implementation
      * @throws SaltStackException
      */
-    public SaltStackClient(String url, SaltStackConnectionFactory connectionFactory)
+    public SaltStackClient(URI url, SaltStackConnectionFactory connectionFactory)
             throws SaltStackException {
```

Comment:
Looks like `SaltStackException` is not thrown anymore, please cleanup method signature and any relevant callers.